File: src/main/scala/org/biodatagraphdb/alsdb/value/NeurobankEventTimepoint.scala
package org.biodatagraphdb.alsdb.value

import org.apache.commons.csv.CSVRecord

case class NeurobankEventTimepoint(
                                    timepointId:String,
                                    subjectId:String,
                                    timepointName:String,
                                    timepoint:Int,
                                    parentTimepointId: String,
                                    timepointInterval:Int,
                                    timepointEventId:Int,
                                    subjectGuid:String
                                    ) {
  val id:String = timepointName
  val subjectTuple:Tuple2[String,String] = new Tuple2(subjectId,subjectGuid)
  val timepointTuple:Tuple2[String,String] = new Tuple2(timepointId, timepointName)
}
object NeurobankEventTimepoint extends ValueTrait {
  def parseCSVRecord(record:CSVRecord):NeurobankEventTimepoint = {
    new NeurobankEventTimepoint(
      record.get("timepoint_id"),
      record.get("subject_id"),
      record.get("timepoint_name"),
      validIntegerString(record.get("timepoint")),
      record.get("parent_timepoint_id"),
      validIntegerString(record.get("timepoint_interval")),
      record.get("timepoint_event_id").toInt,
      record.get("subject_guid")
    )
  }

}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/NeurobankEventTimepoint.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

data class NeurobankEventTimepoint(
        val timepointId: String,
        val subjectId: String,
        val timepointName: String,
        val timepoint: Int,
        val parentTimepointId: String,
        val timepointInterval: Int,
        val timepointEventId: Int,
        val subjectGuid: String
) {
    val id: String = timepointName
    val subjectIdPair = Pair(subjectId, subjectGuid)
    val timepointIdPair = Pair(timepointId, timepointName)

    companion object : AlsdbModel {
        fun parseCSVRecord(record: CSVRecord): NeurobankEventTimepoint =
                NeurobankEventTimepoint(
                        record.get("timepoint_id"),
                        record.get("subject_id"),
                        record.get("timepoint_name"),
                        parseValidIntegerFromString(record.get("timepoint")),
                        record.get("parent_timepoint_id"),
                        parseValidIntegerFromString(record.get("timepoint_interval")),
                        record.get("timepoint_event_id").toInt(),
                        record.get("subject_guid")
                )
    }
}